src/chart-addons/stacked-legend.js

```diff
@@ -392,38 +462,40 @@ function legendState(state, useMap = true) {
     assert(Array.isArray(state.domain))
     assert(state.domain.length === 2)
     assert(typeof state.domain[0] === "number")
-    const range_diff = (state.domain[1] - state.domain[0]) / state.range.length
-    let curr_val = state.domain[0]
-    const domain_labels = state.range.map((item, index, range_array) => {
-      const prev_val = curr_val
-      curr_val += range_diff
-      if (index === 0) {
-        return `< ${formatNumber(curr_val)}`
-      } else if (index === range_array.length - 1) {
-        return `>= ${formatNumber(prev_val)}`
-      } else {
-        return `[${formatNumber(prev_val)}, ${formatNumber(curr_val)})`
-      }
-    })
     return {
-      type: "nominal",
+      type: "gradient",
       title,
+      locked,
       open,
       position,
-      domain: domain_labels,
+      domain: state.domain,
       range: state.range
     }
   } else if (is_quantitative_type) {
     const { scale = {} } = state
     const domain = scale.domain || state.domain
     const range = scale.range || state.range
     const min_size = Math.min(domain.length, range.length)
+    // TODO(croot): may want to consider filling out the auto-gradient here
+    // using a max-number of stops. The best way to do this will most likely
+    // be to create a d3 scale of similar type, and generate new ranges from
+    // it at auto-gradient stops. Unforunately the legendables library will only
+    // auto fill the domain as long as the domain is only of size 2 and the range
+    // has more than 2 values. Instead, we need to handle the case where there's
+    // an implied gradient between multiple values in domain/range where there is
+    // an equal number of domains as ranges. So if we have a domain of [0,100] and
+    // a range of ["blue", "red"], a domain value of 50 would produce purple.
+    // This is currently not accounted for in ledgendables gradient, and therefore
+    // can result in a loss of legend color mapping. So, we would need to auto fill
+    // in some more domain/range stops so the legendables library has data to
+    // display. Because of the varied nature of the quantitative code (linear, pow, log, etc),
+    // it would be best to generate a d3 scale to generate values at linear stops.
```

Comment:
@vastcharade when I was trying to meet the "increments of 5" requirement that we determined didn't make sense, this was essentially may approach and led to this nifty little `arraySample` util:

https://codesandbox.io/s/evenly-distributed-array-sample-4ig6o1?file=/src/index.ts

Reply:
thanks, good to now. however, that util only works for linear functions. To cover all our bases here, we would need to support non-linear too, such as pow, log, sqrt functions. d3 would be our best place for that.